slices/slices.go

```diff
@@ -256,3 +256,25 @@ func Grow[S ~[]E, E any](s S, n int) S {
 func Clip[S ~[]E, E any](s S) S {
 	return s[:len(s):len(s)]
 }
+
+// Difference finds elements that exist in slice 1 but not in slice 2, and there may be duplicate elements in the returned slice.
+func Difference[E comparable](s1, s2 []E) []E {
+	var result []E
+	s2len := len(s2)
+	s1len := len(s1)
+	if s2len == 0 {
+		return s1
+	}
+	s2Map := make(map[E]bool, s2len)
```

Comment:
Maybe we should use `map[E]struct{}`?

```suggestion
	s2Map := make(map[E]struct{}, s2len)
```